tachyon/crypto/commitments/merkle_tree/field_merkle_tree/field_merkle_tree_mmcs.h

```diff
@@ -76,27 +74,30 @@ class FieldMerkleTreeMMCS final
   };
 
   [[nodiscard]] bool DoCommit(std::vector<math::RowMajorMatrix<F>>&& matrices,
-                              Commitment* result) {
-    field_merkle_tree_ =
+                              Commitment* commitment, ProverData* prover_data) {
+    *prover_data =
```

Comment:
The purpose of FieldMerkleTreeMMCS seems ambiguous now that field_merkle_tree_ has been separated into prover_data. Despite its name, it only contains information such as the hasher and compressor, which seems a bit odd to me.

Reply:
Similar to `GetMatrices`, I followed the implementation from Plonky3, though this makes a bit more sense to me. This mmcs hold the functions to run computations, but simply does not carry personal members. As said before as well, I think its good to keep it this ways since it follows Plonky3.